test-clients/src/main/java/com/hedera/services/bdd/suites/contract/precompile/PrngPrecompileSuite.java

```diff
@@ -0,0 +1,230 @@
+    private static final String bob = "bob";
+
+    public static void main(String... args) {
+        new PrngPrecompileSuite().runSuiteSync();
+    }
+
+    @Override
+    public boolean canRunConcurrent() {
+        return false;
+    }
+
+    @Override
+    public List<HapiApiSpec> getSpecsInSuite() {
+        return allOf(positiveSpecs(), negativeSpecs());
+    }
+
+    List<HapiApiSpec> negativeSpecs() {
+        return List.of(
+                functionCallWithLessThanFourBytesFailsGracefully(),
+                nonSupportedAbiCallGracefullyFails(),
+                invalidLargeInputFails(),
+                emptyInputCallFails());
+    }
+
+    List<HapiApiSpec> positiveSpecs() {
+        return List.of(prngPrecompileHappyPathWorks());
+    }
+
+    private HapiApiSpec emptyInputCallFails() {
+        final var prng = THE_GRACEFULLY_FAILING_PRNG_CONTRACT;
+        final var emptyInputCall = "emptyInputCall";
+        return defaultHapiSpec("emptyInputCallFails")
+                .given(cryptoCreate(bob), uploadInitCode(prng), contractCreate(prng))
+                .when(
+                        sourcing(
+                                () ->
+                                        contractCall(prng, "performEmptyInputCall")
+                                                .gas(GAS_TO_OFFER)
+                                                .payingWith(bob)
+                                                .via(emptyInputCall)
+                                                .hasKnownStatus(CONTRACT_REVERT_EXECUTED)
+                                                .logged()))
+                .then(
+                        getTxnRecord(emptyInputCall)
+                                .andAllChildRecords()
+                                .logged()
+                                .saveTxnRecordToRegistry(emptyInputCall),
+                        withOpContext(
+                                (spec, ignore) -> {
+                                    final var gasUsed =
+                                            spec.registry()
+                                                    .getTransactionRecord(emptyInputCall)
+                                                    .getContractCallResult()
+                                                    .getGasUsed();
+                                    assertEquals(394213, gasUsed);
+                                }));
+    }
+
+    private HapiApiSpec invalidLargeInputFails() {
+        final var prng = THE_GRACEFULLY_FAILING_PRNG_CONTRACT;
+        final var largeInputCall = "largeInputCall";
+        return defaultHapiSpec("invalidLargeInputFails")
+                .given(cryptoCreate(bob), uploadInitCode(prng), contractCreate(prng))
+                .when(
+                        sourcing(
+                                () ->
+                                        contractCall(prng, "performLargeInputCall")
```

Comment:
There is no extra data passed in. Perhaps `withExplicitParams` with a very large byte array?

Reply:
Thank you ! Fixed to use `withExplicitParams`